barcalendar.py

```diff
@@ -245,6 +249,51 @@ def column_marker(self, column):
     def write(self):
         self.epilog()
 
+def get_defaults_from_tutor():
+    """
+    Fetches default configurations from tutor repository.
+    Returns:
+        object: Default configurations as Python object
+    """
+    url = "https://raw.githubusercontent.com/overhangio/tutor/master/tutor/templates/config/defaults.yml"
+    while True:
+        try:
+            resp = requests.get(url, timeout=10)
+        except requests.RequestException as exc:
+            print(f"Couldn't fetch {url}: {exc}")
+            return None
+        if resp.status_code == 429:
+            wait = int(resp.headers.get("Retry-After", 10))
+            time.sleep(wait + 1)
+        else:
+            break
+
+    if resp.status_code == 200:
+        return safe_load(resp.text)
+    return None
+
+def parse_version_number(line):
+    """
+    Get version number in line from YAML file.
+    Note that this only captures major and minor version (not patch number). 
+    e.g. "docker.io/elasticsearch:7.17.9" -> "7.17"
+    """
+    match = re.search(r'(?P<version_number>\d+(\.\d+)?)', line)
+    if match is not None:
+        version_number = match.group("version_number")
+        return version_number
+    return None
+
+def parse_version_name(line):
+    """
+    Get openedx version name in line from YAML file.
+    e.g. "open-release/palm.1" -> "Palm"
+    """
+    match = re.search(r'(?P<version_name>[A-Za-z]+)[\.\d+]*$', line)
```

Comment:
I'm not sure how this setting is managed.  Sometimes its value is `open-release/palm.master`.  Perhaps the best approach is to use:
```
match = re.search(r'/(?P<version_name>[A-Za-z]+)\.', line)
```

Reply:
Modified it to 

``` Python
match = re.search(r'(?P<version_name>[A-Za-z]+)(\.\w*)*$', line)
``` 
to also match the following cases:
- `"open-release/palm.1"`
- `"open-release/palm.1.2.3"`
- `"open-release/palm.master"`
- `"open-release/palm.master.1.2.3"`
- `"open-release/palm."`

It is a bit redundant. Looking at [this page,](https://docs.openedx.org/en/latest/community/release_notes/named_release_branches_and_tags.html) it shouldn't deviate from the two formats we considered here. 

Happy to go with the simpler approach.

Reply:
To me, it's simpler to say, "the word between the slash and the dot."  Your regex is, "the word followed by dots and words at the end of the line," which seems more confusing.

Reply:
When you phrase it that way, I completely agree. I've made the change.